data-tidy.qmd

```diff
@@ -569,8 +569,10 @@ Since you don't know how to work with this sort of data yet, you'll want to foll
 
 ```{r}
 df |> 
-  group_by(id, name) |> 
-  summarize(n = n(), .groups = "drop") |> 
+  summarize(
+    .by = c(id, name),
+    n = n(),
+  ) |> 
   filter(n > 1L) 
```

Comment:
In terms of teaching, is it easier to explain that you go from:

```r
df %>% summarise(x = mean(x))
```

to:

```r
df %>% summarise(x = mean(x), .by = c(g1, g2))
```

or:

```r
df %>% summarise(.by = c(g1, g2), x = mean(x))
```

?

The latter feels somewhat unnatural for beginners who aren't used to the idea that the placement of named arguments is arbitrary. They are also likely using simple `summarise()` expressions where it is unlikely that the `.by` call will get "lost"

Reply:
Agreed